### DISCUSSION TITLE
Image files generated during build are deleted

### DISCUSSION THREAD

**Discussion by PhilippHandle** [tighten/jigsaw#567] [Q&A]
i implemented this https://gist.github.com/ryanscherler/f14976db7f1c0e5103b741a4a3b5858e and by the look the file gets delete after.

if i add an exit before the return the file stays there, but with out it it gets deleted.
if i change the code to write to the source folder, i have to run the build a second time.

Any tips?

**Comment by bakerkretzmar** [comment]
Whoa, that looks really cool! I'm not positive what's causing the file to get deleted but I'll take another look.

Off the top of my head, I wonder if you could solve this by splitting it into two listeners, one on `beforeBuild` that sets up your `resizeImage()` method and then just populates a global list of images and paths that need to be transformed, and then one on `afterBuild` to actually make the transformations and move the files?
